Improve GC Heap Inspector

ast-hugger opened the pull request:
#### 5d27374d1d1494e909102a0deefe4a19252b4cef
<pre>
Improve GC Heap Inspector
<a href="https://bugs.webkit.org/show_bug.cgi?id=295656">https://bugs.webkit.org/show_bug.cgi?id=295656</a>
<a href="https://rdar.apple.com/155461683">rdar://155461683</a>

Reviewed by Ryan Reno, Justin Michaud, and Yusuke Suzuki.

This patch adds new functionality and documentation to GC Heap Inspector,
allowing it to display dominator information to identify certain kinds of
memory leaks, and to navigate the object graph.

* Tools/GCHeapInspector/README.md: Added.
* Tools/GCHeapInspector/gc-heap-inspector.html:
* Tools/GCHeapInspector/script/interface.js:

Canonical link: <a href="https://commits.webkit.org/297400@main">https://commits.webkit.org/297400@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4ec542d6fe2b314bce00343a6e3c552961a94bd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111206 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30872 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21354 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117237 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61474 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/113168 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31553 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39454 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84529 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114153 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25204 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100146 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64975 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24546 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18291 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61057 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94582 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18356 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/120297 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38255 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28425 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93458 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38631 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96423 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93283 "Found 2 new API test failures: /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy, /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23848 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38374 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16164 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/34240 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38144 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37809 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41142 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39511 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rreno commented on Tools/GCHeapInspector/script/interface.js:
```diff
@@ -232,43 +329,45 @@ class HeapSnapshotInspector
         header = document.createElement('h1');
-        header.textContent = 'All paths to…'
-        this.allPathsContainer.appendChild(header);
+        header.textContent = 'Tools'
```
Can this be called "Node Details" like in the README? Or maybe just Details?

ast-hugger replied:
Yep, Details sounds good.

rreno reviewed (approved):
This is great! r=me with comment.

justinmichaud reviewed (approved):
This is awesome!

ast-hugger committed:
Improve GC Heap Inspector
https://bugs.webkit.org/show_bug.cgi?id=295656
rdar://155461683

Reviewed by Ryan Reno, Justin Michaud, and Yusuke Suzuki.

This patch adds new functionality and documentation to GC Heap Inspector,
allowing it to display dominator information to identify certain kinds of
memory leaks, and to navigate the object graph.

* Tools/GCHeapInspector/README.md: Added.
* Tools/GCHeapInspector/gc-heap-inspector.html:
* Tools/GCHeapInspector/script/interface.js:

Canonical link: https://commits.webkit.org/297400@main
